src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmDirectValueAnnotationsManager.cs

```diff
@@ -230,96 +219,60 @@ private static void SetAnnotation(IEnumerable<IEdmDirectValueAnnotation> immutab
             transientAnnotations = annotationsList.Add(newAnnotation);
         }
 
-        private static IEdmDirectValueAnnotation FindTransientAnnotation(object transientAnnotations, string namespaceName, string localName)
+        private static IEdmDirectValueAnnotation FindTransientAnnotation(VersioningList<IEdmDirectValueAnnotation> transientAnnotations, string namespaceName, string localName)
         {
             if (transientAnnotations != null)
             {
-                IEdmDirectValueAnnotation singleAnnotation = transientAnnotations as IEdmDirectValueAnnotation;
-                if (singleAnnotation != null)
+                VersioningList<IEdmDirectValueAnnotation> annotationsList = transientAnnotations;

```

Comment:
Do we need to declare `annotationsList` yet we can loop over `transientAnnotations`?